server/api/commands.go

```diff
@@ -53,20 +44,29 @@ func LoadTest(w stdio.Writer, state State, waitForCancel func(), throttle string
 		io.ReplyEither(w, err)
 		return result
 	}
-	var targets t.Targets = []t.Target{
-		{"POST", url, reqBody, conc, float64(qps)},
-	}
-	work := targets.GetWork(state.ApmServer().Url(), &t.Config{
-		MaxRequests: math.MaxInt32,
-		// should match the one in apm-server
-		RequestTimeout:     30,
+
+	cfg := &target.Config{
+		Concurrent:     conc,
+		Qps:            float64(qps),
+		MaxRequests:    math.MaxInt32,
+		RequestTimeout: 30,
+		Endpoint:       "/intake/v2/events",
+		BodyConfig: &target.BodyConfig{
+			NumErrors:       numErrors,
+			NumTransactions: numTransactions,
+			NumSpans:        numSpans,
+			NumFrames:       numFrames,
+		},
 		DisableCompression: false,
-	})[0]
+	}
+	t := target.NewTarget(state.ApmServer().Url(), "POST", cfg)
+	work := t.GetWork()
+
 	docsBefore := state.ElasticSearch().Count()
 	start := time.Now()
 	go work.Run()
 	io.ReplyNL(w, io.Grey+fmt.Sprintf("started new work, payload size is %s...",
-		byteCountDecimal(int64(len(reqBody)))))
+		byteCountDecimal(int64(len(t.Body)))))
```

Comment:
i think would be cool to move L68 up before the call to `t.GetWork` to get the uncompressed size and rephrase the message to `"started new work, request body uncompressed size is ..."`

and then again, after calling `t.GetWork` log again the compressed size
`io.ReplyNL(w, io.Grey+fmt.Sprintf("compressed size is %s...", byteCountDecimal(int64(len(reqBody)))))`